app/models/cart.rb

```diff
@@ -4,13 +4,19 @@ class Cart
   def initialize(initial_contents)
     @contents = initial_contents || {}
   end
-
+  
   def add_beer(beer, quantity)
-    if contents.has_key?(beer.id)
-      contents[beer.id][:quantity] += quantity
-      contents[beer.id][:beer_price] = beer.price * contents[beer.id][:quantity]
+    if contents.has_key?(beer.id.to_s)
+      contents[beer.id.to_s]["quantity"] = contents[beer.id.to_s]["quantity"].to_i + quantity.to_i
+      contents[beer.id.to_s]["beer_price"] = beer.price * contents[beer.id.to_s]["quantity"].to_i
     else
-      contents[beer.id] = {beer_name: beer.name, beer_price: beer.price, quantity: quantity} 
+      contents[beer.id.to_s] = {"beer_name" => beer.name, "beer_price" => beer.price, "quantity" => quantity}
     end
   end
+  
+  def total
+    total = 0
+    contents.each { |k, v| total += v["beer_price"].to_i }
```

Comment:
Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.